frappe/utils/data.py

```diff
@@ -939,6 +939,15 @@ def flt(s: NumericType | str, precision: int | None = None) -> float:
 	>>> flt("a")
 	0.0
 	"""
+
+	#Get current decimal string and convert it to usual one.
+	number_format = frappe.db.get_default("number_format") or "#,###.##"
+	decimal_str, comma_str, number_format_precision = get_number_format_info(number_format)
```

Comment:
Number format is for visual representation, not machine representation. So this should be just handled in client-side JS code for controls by storing computer parsable representation instead of user-facing one. AFAIK that's the intended design, let's not add number format checks in low-level functions.  


The problem here is that `Data` field type is used for storing numbers instead of `Float` or equivalent numeric field. That's only